src/quick-lint-js/parse.h

```diff
@@ -1441,12 +1441,41 @@ class parser {
     QLJS_WARNING_IGNORE_GCC("-Wshadow-local")
 
     std::optional<identifier> last_ident;
+    const char8 *async_token_begin = nullptr;
     function_attributes method_attributes = function_attributes::normal;
+    bool async_static = false;
 
     auto parse_and_visit_field_or_method_impl =
-        [this, &v](std::optional<identifier> property_name,
-                   source_code_span property_name_span,
-                   function_attributes method_attributes) -> void {
+        [this, &v, &async_static, &async_token_begin](
+            std::optional<identifier> property_name,
+            source_code_span property_name_span,
+            function_attributes method_attributes) -> void {
+      if (async_static) {
+        if (this->peek().type == token_type::star) {
+          method_attributes = function_attributes::async_generator;
+          this->skip();
+        }
+        switch (this->peek().type) {
+        QLJS_CASE_RESERVED_KEYWORD_EXCEPT_FUNCTION:
+        QLJS_CASE_CONTEXTUAL_KEYWORD:
+        case token_type::identifier:
+        case token_type::private_identifier:
+        case token_type::reserved_keyword_with_escape_sequence:
+          // async static method() {}             // Invalid
+          // async static *myAsyncGenerator() {}  // Invalid
+          this->error_reporter_->report(error_async_static_method{
+              .async_static =
+                  source_code_span(async_token_begin, property_name_span.end()),
+          });
+          property_name = this->peek().identifier_name();
+          property_name_span = property_name->span();
+          this->skip();
+          break;
+        default:
```

Comment:
Maybe this deserves a comment?
```suggestion
        // async static() {}
        default:
```